extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -102,22 +114,26 @@ export class ConsentUI {
       // No prompt UI specified, nothing to do
       return;
     }
-    toggle(this.parent_, true);
+    toggle(dev().assertElement(this.parent_), true);
     const {classList} = this.parent_;
     classList.add('amp-active');
     classList.remove('amp-hidden');
     // Add to fixed layer
     this.baseInstance_.getViewport().addToFixedLayer(this.parent_);
-    toggle(this.ui_, true);
     if (this.isCreatedIframe_) {
-      // TODO: Apply placeholder and hide iframe
-      insertAfterOrAtStart(this.parent_, this.ui_, null);
-    }
-    if (!this.isPostPrompt_ && !this.isCreatedIframe_) {
-      // scheduleLayout is required everytime because some AMP element may
-      // get un laid out after toggle display (#unlayoutOnPause)
-      // for example <amp-iframe>
-      this.baseInstance_.scheduleLayout(this.ui_);
+      this.iframeReady_ = new Deferred();
+      this.loadIframe_();
```

Comment:
How about this:
```js
this.loadIframe_().then(() => {this.showIframe_()}
```

```
loadIframe() {
  ...
  this.ui_.readyDeferred = new Deferred();
  return this.ui_.readyDeferred.promise;
}

handleIframeMessages_(event) {
  ...
  this.ui_.readyDeferred.resolve();
}
```

This way, we can reset this.ui_  without worrying about the iframeReady state.

Reply:
done